SpatialGDK/Source/SpatialGDK/Public/Utils/EntityFactory.h

```diff
@@ -25,12 +25,39 @@ struct RPCsOnEntityCreation;
 using FRPCsOnEntityCreationMap = TMap<TWeakObjectPtr<const UObject>, RPCsOnEntityCreation, FDefaultSetAllocator,
 									  TWeakObjectPtrMapKeyFuncs<TWeakObjectPtr<const UObject>, SpatialGDK::RPCsOnEntityCreation, false>>;
 
+struct EntityComponents
+{
+	TMap<Worker_ComponentId, TUniquePtr<DataComponent>> ModifiableComponents;
+	TMap<Worker_ComponentId, FWorkerComponentData> ComponentsToDelegateToAuthoritativeWorker;
+	TArray<FWorkerComponentData> ComponentDatas;
+
+	// PRCOMMENT: Hmm, will this incur overhead? Probably...
```

Comment:
Comment for PR purposes: I don't like this, I think we are copying the data when we maybe could avoid that...